src/dmd/e2ir.d

```diff
@@ -669,8 +674,19 @@ elem *array_toPtr(Type t, elem *e)
             }
             else if (e.Eoper == OPpair)
             {
-                e.Eoper = OPcomma;
-                e.Ety = TYnptr;
+                if (el_sideeffect(e.EV.E1))
```

Comment:
Any chance to add a test where this is actually true?

Reply:
When this condition is true this function uses the same behavior as prior to this PR.